lib/include/ultrahdr/jpegdecoderhelper.h

```diff
@@ -38,139 +38,145 @@ extern "C" {
 namespace ultrahdr {
 
 // constraint on max width and max height is only due to device alloc constraints
-// Can tune these values basing on the target device
+// can tune these values basing on the target device
 static const int kMaxWidth = 8192;
 static const int kMaxHeight = 8192;
 
+/*!\brief List of supported operations */
 typedef enum {
-  PARSE_ONLY = 0,          // Dont decode. Parse for dimensions, EXIF, ICC, XMP
-  DECODE_TO_RGBA = 1,      // Parse and decode to rgba
-  DECODE_TO_YCBCR = 2,     // Parse and decode to YCBCR or Grayscale
-                           // if input has 1 channel, decode to Grayscale
-                           // if input has 3 channels, decode to YCBCR
-  DECODE_TO_GAIN_MAP = 3,  // parse and decode gain map.
-                           // if input has 1 channel, decode to Grayscale
-                           // if input has 3 channels, decode to RGBA
+  PARSE_STREAM = (1 << 0),   /**< Parse jpeg header, APPn markers (Exif, Icc, Xmp, Iso) */
+  DECODE_STREAM = (1 << 16), /**< Single channel images are decoded to YCbCr Color Space and multi
+                              channel images are decoded to RGB Color Space */
+  DECODE_TO_YCBCR_CS = (1 << 17), /**< Decode image to YCbCr Color Space  */
+  DECODE_TO_RGB_CS = (1 << 18),   /**< Decode image to RGB Color Space  */
 } decode_mode_t;
 
-/*
- * Encapsulates a converter from JPEG to raw image (YUV420planer or grey-scale) format.
- * This class is not thread-safe.
- */
+/*!\brief Encapsulates a converter from JPEG to raw image format. This class is not thread-safe */
 class JpegDecoderHelper {
  public:
-  JpegDecoderHelper();
-  ~JpegDecoderHelper();
-  /*
-   * Decompresses JPEG image to raw image (YUV420planer, grey-scale or RGBA) format. After
-   * calling this method, call getDecompressedImage() to get the image.
-   * Returns false if decompressing the image fails.
-   */
-  bool decompressImage(const void* image, int length, decode_mode_t decodeTo = DECODE_TO_YCBCR);
-  /*
-   * Returns the decompressed raw image buffer pointer. This method must be called only after
```

Comment:
For all the similar notes, I think they are necessary. Could you add those to the new methods' comments?

Reply:
I mean we need to keep "This method must be called only after ..." note

Reply:
They are present dichen, I did not remove. I move them to one place.

Lines 102-103
/*! Below public methods are only effective if a call to decompressImage() is made and it returned
   * true. */
<CODE>
Lines 114-115
  /*! Below public methods are only effective if a call to parseImage() or decompressImage() is made
   * and it returned true. */